backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/picking/job/service/PickingJobService.java

```diff
@@ -341,6 +342,17 @@ public void unassignAllByUserId(@NonNull final UserId userId)
 		});
 	}
 
+	public boolean clearAssignmentsForSlot(@NonNull final PickingSlotId slotId, final boolean abortOngoingPickingJobs)
+	{
+		final List<PickingJob> pickingJobs = pickingJobRepository.getDraftedByPickingSlotId(slotId, pickingJobLoaderSupportingServicesFactory.createLoaderSupportingServices());
+		if (pickingJobs.isEmpty())
+		{
+			return true;
+		}
+
+		return pickingJobs.stream().allMatch(job -> removePickingSlotAssignment(job, abortOngoingPickingJobs));
```

Comment:
calling allMatch with removePickingSlotAssignment feels a bit weird because
* allMatch is supposed to get a predicate that is not changing data
* it might be that allMatch will stop at first "false" because there is no point in going on. not sure if that's desired

Reply:
that's exactly what I want, it will be rolled back anyways